src/ekat/kokkos/ekat_subview_utils.hpp

```diff
@@ -316,6 +314,177 @@ subview_1(const ViewLR<ST******,Props...>& v,
       v.impl_track(),vm);
 }
 
+// ================ Multi-sliced Subviews ======================= //
+// e.g., instead of a single-entry slice like v(:, 42, :), we slice over a range
+// of values, as in v(:, 27:42, :)
+// this means that the subview has the same rank as the source view
+
+// --- Rank1 multi-slice --- //
+template <typename ST, typename... Props>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<ViewLS<ST*, Props...>>
+subview(const ViewLR<ST*, Props...>& v,
+        const Kokkos::pair<int, int> &kp0,
+        const int idim) {
+  assert(v.data() != nullptr);
+  assert(idim == 0);
+  // NOTE: the final comparison is originally int <= long unsigned int
+  // the cast silences a warning, but may be unnecessary
+  assert(kp0.first >= 0 && kp0.first < kp0.second
+         && kp0.second <= v.extent_int(idim));
+  return Unmanaged<ViewLS<ST*,Props...>>(Kokkos::subview(v, kp0));
+
+}
+
+// --- Rank2 multi-slice --- //
+template <typename ST, typename... Props>
+KOKKOS_INLINE_FUNCTION
+Unmanaged<ViewLS<ST**, Props...>>
+subview(const ViewLR<ST**, Props...>& v,
+        const Kokkos::pair<int, int> &kp0,
+        const int idim) {
+  assert(v.data() != nullptr);
+  assert(idim >= 0 && idim <= v.rank);
+  // NOTE: the final comparison is originally int <= long unsigned int
+  // the cast silences a warning, but may be unnecessary
+  assert(kp0.first >= 0 && kp0.first < kp0.second
+         && kp0.second <= v.extent_int(idim));
+  if (idim == 0) {
```

Comment:
If you don't like these conditionals, we could bake the index in the function name. Someone like `subview_0` for subviewing along the host index vs `subview_1` for sunburn l subviewing asking the second. I _think_ in most (all?) use cases we will know which dimension we're subviewing anyways. @tcclevenger what do you think?

Reply:
My only issue is this would end up creating a lot of function, right? `subview_1()` would need 6 versions, `subview_2()` would have 5, etc. And then in the SCREAM-side PR we would end up having a long `switch(idim) case 1:, case 2:, ...` chain.

Reply:
Unless there's an advantage to changing, I suppose I'm in favor of leaving it as-is, but could certainly be convinced to go with the opposite case

Reply:
Unless @bartgol  has push back I'd say keep this how it is, and then later we may rethink all the `subview` routines.